projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts

```diff
@@ -271,8 +279,10 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
     }
   }
 
-  private updateData(): void {
-    this.drawLegend();
+  private updateData(withLegend: boolean = true): void {
```

Comment:
boolean flags to change behavior are a code smell. Why were we able to always draw legend up until now, and now need to configure it?

Reply:
So the current behaviour was to draw everything when series changes. In this current case I do not want to draw the legend again as we need every state etc. for the legend.

Reply:
Commented on this below - we don't want to hold state in the legend in that way.